server/api/crud/alerts.py

```diff
@@ -266,5 +262,23 @@ def reset_alert(self, session: sqlalchemy.orm.Session, project: str, name: str):
     def _delete_notifications(self, alert: mlrun.common.schemas.AlertConfig):
         for notification in alert.notifications:
             server.api.api.utils.delete_notification_params_secret(
-                alert.project, notification
+                alert.project, notification.notification
+            )
+
+    def _validate_and_mask_notificatsions(self, alert_data):
+        notifications = [
+            mlrun.common.schemas.notification.Notification(**notification.to_dict())
+            for notification in server.api.api.utils.validate_and_mask_notification_list(
+                alert_data.get_notifications(), None, alert_data.project
+            )
+        ]
+        cooldowns = [
+            notification.cooldown_period for notification in alert_data.notifications
+        ]
+
+        alert_data.notifications = [
+            mlrun.common.schemas.alert.AlertNotification(
+                **{"cooldown_period": cooldown, "notification": notification}
             )
```

Comment:
```suggestion
            mlrun.common.schemas.alert.AlertNotification(
                cooldown_period=cooldown, notification=notification
            )
```
No need to create a dict and expand it here.